plugins/Jobs.cpp

```diff
@@ -201,147 +240,208 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
+            const string& safeFirstRun = !SContains(job, "firstRun") || job["firstRun"].empty() ? SCURRENT_TIMESTAMP() : SQ(job["firstRun"]);
 
-        // If no priority set, set it
-        int64_t priority = request.isSet("priority") ? request.calc("priority") : JOBS_DEFAULT_PRIORITY;
+            // If no data was provided, use an empty object
+            const string& safeData = !SContains(job, "data") || job["data"].empty() ? SQ("{}") : SQ(job["data"]);
 
-        // Validate that the parentJobID exists and is in the right state if one was passed.
-        int64_t parentJobID = request.calc64("parentJobID");
-        if (parentJobID) {
-            auto parentState = db.read("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
-            if (parentState.empty()) {
-                throw "404 parentJobID does not exist";
-            } else if (!SIEquals(parentState, "RUNNING") && !SIEquals(parentState, "PAUSED")) {
-                SWARN("Trying to create child job with parent jobID#" << parentJobID << ", but parent isn't RUNNING or PAUSED (" << parentState << ")");
-                throw "405 Can only create child job when parent is RUNNING or PAUSED";
+            // If a repeat is provided, validate it
+            if (SContains(job, "repeat")) {
+                if (job["repeat"].empty()) {
+                    SWARN("Repeat is set in CreateJob, but is set to the empty string. Job Name: "
+                          << job["name"] << ", removing attribute.");
+                    job.erase("repeat");
+                } else if (!_validateRepeat(job["repeat"])) {
+                    throw "402 Malformed repeat";
+                }
             }
-        }
 
-        // We'd initially intended for any value to be allowable here, but for
-        // performance reasons, we currently will only allow specific values to
-        // try and keep queries fast. If you pass an invalid value, we'll throw
-        // here so that the caller can know that he did something wrong rather
-        // than having his job sit unprocessed in the queue forever. Hopefully
-        // we can remove this restriction in the future.
-        if (priority != 0 && priority != 500 && priority != 1000) {
-            throw "402 Invalid priority value";
-        }
-
-        // Are we creating a new job, or updating an existing job?
-        if (updateJobID) {
-            // Update the existing job.
-            if(!db.write("UPDATE JOBS SET "
-                           "repeat   = " + SQ(SToUpper(request["repeat"])) + ", " +
-                           "data     = JSON_PATCH(data, " + safeData + "), " +
-                           "priority = " + SQ(priority) + " " +
-                         "WHERE jobID = " + SQ(updateJobID) + ";"))
-            {
-                throw "502 update query failed";
+            // If no priority set, set it
+            int64_t priority = SContains(job, "priority") ? SToInt(job["priority"]) : JOBS_DEFAULT_PRIORITY;
```

Comment:
Can you use `job.isSet("priority")` and `job.calc["priority"]` instead of ` SContains(job, "priority")` and  `SToInt(job["priority"])`?